### PR TITLE
Vendors improve

### PR DESCRIPTION
**Pull request by goose4429**
## About The Pull Request

This PR improves loadout system. 
- It now can use several categories with independent points for each.
- New universal/choice categories can be easily added or altered

Feel free to ping me on discord **qwerty4429**

## Why It's Good For The Game

- Code improve good
- Functionality expansion good

## Details

- Outfit stuff now uses normal list instead of list-of-lists and bitfields. Pouches are now non-hardcoded.
- Any numbers can be put into choice cats, as well as new cats can be added.
- Class vendors properly utilize several point categories. Add points through card_ids.dm
- Other code refactors
- Other small non-balance changes

## Changelog
:cl:
code: Refactored vendor code for loadouts
code: Vendors now support several point categories
/:cl:

### CODE REVIEW

**Comment by QualityVan** [comment]
No but really you should probably bundle the reagent/recipe changes separate from the vendor changes.

File: code/game/objects/machinery/vending/new_marine_vendors.dm [hunk 1]
```diff
@@ -94,18 +93,30 @@
 	. = list()
 
 	var/obj/item/card/id/I = user.get_idcard()
-	.["current_m_points"] = I?.marine_points || 0
 	var/buy_flags = I?.marine_buy_flags || NONE
-
+	var/obj/item/card/id/dogtag/full/ptscheck = new /obj/item/card/id/dogtag/full
```
[the remark this answers is not in the document]
Comment (QualityVan) [reply code/game/objects/machinery/vending/new_marine_vendors.dm:97]: Or you could just type it global? I'm not super confident about that one, so ping a maint if you're curious.

File: code/modules/reagents/reagents/medical.dm [hunk 2]
```diff
@@ -432,8 +427,10 @@
 	L.AdjustUnconscious(-20)
 	L.AdjustStun(-20)
 	L.AdjustParalyzed(-20)
-	L.adjustToxLoss(effect_str)
 	L.hallucination = max(0, L.hallucination - 10)
+	L.adjustToxLoss(effect_str)
+	L.reagents.add_reagent(/datum/reagent/toxin, 2)
```
[the remark this answers is not in the document]
Comment (goose4429) [reply code/modules/reagents/reagents/medical.dm:432]: I'm just using it so you cant 'just' use syna. 20 ticks of syna will give you 40 toxin, and you can't eat dylo when you have syna in blood. But yeah, pretty weak.

File: code/game/objects/machinery/vending/new_marine_vendors.dm (hunk 1 above)
[the remark this answers is not in the document]
Comment (goose4429) [reply code/game/objects/machinery/vending/new_marine_vendors.dm:97]: Will do machine thing you've mentioned

**Comment by TiviPlus** [comment]
> If you ask me to atomize, I'll kill you.

https://tenor.com/view/cat-door-kitty-open-feline-gif-20348122

**Comment by Lumipharon** [comment]
This change cucks medics. No reason to drop their point limit and split injectors out into another category, as there is no particular necessity to buy any of them, and none are powerful enough to warrant people wanting to buy shitloads of them.

Good for normal unga though.

**Comment by Al-1ce** [comment]
This will have conflicts with #10559.

**Comment by goose4429** [comment]
> No reason to drop their point limit

What exactly do you lose from these 5 points?

**Comment by Al-1ce** [comment]
Everything this PR looks to change is already in-place. If I want injectors, I'll take them from the MarineMed and stuff them in an injector pouch. There is no need for this.

**Comment by Lumipharon** [comment]
> > No reason to drop their point limit
> 
> What exactly do you lose from these 5 points?

5 points worth of choice?

What do you gain from splitting out injectors into a separate category and point pool from everything else?
It makes sense for normal marines, not medics.

**Comment by goose4429** [comment]
> 5 points worth of choice?
> What do you gain from splitting out injectors into a separate category and point pool from everything else?
It makes sense for normal marines, not medics.

So it sounds like you don't use them for anything particular. Which is expected, because things which you can buy in vendor for 5 or less points are: Tricord, Nanoblood, MedHud and injectors.

- Nanoblood was increased in price - because it's a cool thing you can abuse with nanites. And also because it's uncraftable T2 chem.
- Tricord is funny, as well as MedHud
- Injectors are moved to separate category
All other things cost 8 or more, and you are unaffected in your choice. In current PR, no thing you can but for points can be bought for less than 8. So these 5 points you are scared to lose are literally unspendable - except for empty hypo. Which is actually a dumb thing to include, you can just take free ones from NTMed+.

Injectors are aimed to serve unique purpose - getting specialized chems or mixes you wouldn't get otherwise
Separate category is because just putting them under normal points would either make you buy less of the stuff you had before, or just ignore new cool part of the game. 

Which you and I want to avoid. Up to a small margin of error, you're better off from this PR as a corpsman. You can still take all MeraDerm stuff you want (except for nanoblood which was deliberately made more expensive). These 5 points are unspendable. Given that you weren't spending 20 points on nanoblood, you get _more_ stuff after PR. And more choices.

Does that sound reasonable to you?

**Comment by goose4429** [comment]
> Everything this PR looks to change is already in-place. If I want injectors, I'll take them from the MarineMed and stuff them in an injector pouch. There is no need for this.

Yeah! Almost all. Except for RR, meraderm, synaptizine, hyronalin, ryetalin, paracetamol. The one injectors which are actually cool. All others (whooping 4 of them) are in MM, yes. Except they aren't, because it's a stronger version of them available in PR.

And I doubt you can take point categories code from MM as well.
Very cool and thoughtful comment, thanks.

**Comment by Lumipharon** [comment]
I mean, I do literally buy nanoblood every round as medic yes.

Splitting new stuff into its own category so people don't "ignore cool new parts of the game" is also very questionable logic.

And you're completely ignoring the fact that yes, a lot of options do cost 8 or more points, so depending on what you're wanting to buy, you are in fact getting fucked by the change.

For me personally, it makes zero difference as I buy the same shit every round and don't even use all my points, but it clearly does limit people's choices purely to force them into buying injectors that they may, or may not actually want.

**Comment by Al-1ce** [comment]
>Almost all. Except for RR, meraderm, synaptizine, hyronalin, ryetalin, paracetamol. The one injectors which are actually cool

Russian red is available in pills from hacked machines. And if you're really desperate you can grind up Red Comrade cigarettes, which are infinite. Or just smoke them which is how they're intended to be used.

MeraDerm is very powerful and I don't believe it should be readily accessible for the average marine. Allowing synaptizine by itself is one thing but adding the two chemicals which negate synaptizine's only downside = infinite stamina.

**Comment by goose4429** [comment]
Will make only loadout, and separate balance to RU

**Pull request merged**